exchanges/graphcache/src/default-storage/index.ts

```diff
@@ -0,0 +1,178 @@
+import { stringifyVariables } from '@urql/core';
+import { SerializedEntries, SerializedRequest, StorageAdapter } from '../types';
+
+const getRequestPromise = <T>(request: IDBRequest<T>): Promise<T> => {
+  return new Promise((resolve, reject) => {
+    request.onerror = () => {
+      reject(request.error);
+    };
+
+    request.onsuccess = () => {
+      resolve(request.result);
+    };
+  });
+};
+
+const getTransactionPromise = (transaction: IDBTransaction): Promise<any> => {
+  return new Promise((resolve, reject) => {
+    transaction.onerror = () => {
+      reject(transaction.error);
+    };
+
+    transaction.oncomplete = resolve;
+  });
+};
+
+export interface StorageOptions {
+  idbName?: string;
+  maxAge?: number;
+}
+
+export interface DefaultStorage extends StorageAdapter {
+  clear(): Promise<any>;
+}
+
+export const makeDefaultStorage = (opts?: StorageOptions): DefaultStorage => {
+  if (!opts) opts = {};
+
+  const DB_NAME = opts.idbName || 'graphcache-v3';
+  const ENTRIES_STORE_NAME = 'entries';
+  const METADATA_STORE_NAME = 'metadata';
+
+  const batch: Record<string, string | undefined> = Object.create(null);
+  const timestamp = Math.floor(new Date().valueOf() / (1000 * 60 * 60 * 24));
+  const maxAge = timestamp - (opts.maxAge || 7);
+
+  const req = indexedDB.open(DB_NAME, 1);
+  const database$ = getRequestPromise(req);
+
+  req.onupgradeneeded = () => {
+    req.result.createObjectStore(ENTRIES_STORE_NAME);
+    req.result.createObjectStore(METADATA_STORE_NAME);
+  };
+
+  const serializeBatch = (): string => {
+    let data = '';
+    for (const key in batch) {
+      const value = batch[key];
+      data += `${stringifyVariables(key)}:${
+        value !== undefined ? stringifyVariables(value) : 'null'
+      },`;
+    }
+
+    return data;
+  };
+
+  const deserializeBatch = (input: string) => {
+    try {
+      return JSON.parse(`{${input.slice(0, -1)}}`);
+    } catch (_error) {
+      return {};
+    }
+  };
+
+  return {
+    clear() {
+      return database$.then(database => {
+        const transaction = database.transaction(
+          METADATA_STORE_NAME,
+          'readwrite'
+        );
+        transaction.objectStore(METADATA_STORE_NAME).clear();
+        transaction.objectStore(ENTRIES_STORE_NAME).clear();
+        return getTransactionPromise(transaction);
+      });
+    },
+
+    readMetadata(): Promise<SerializedRequest[]> {
+      return database$.then(
+        database => {
+          return getRequestPromise<SerializedRequest[]>(
+            database
+              .transaction(METADATA_STORE_NAME, 'readonly')
+              .objectStore(METADATA_STORE_NAME)
+              .get(METADATA_STORE_NAME)
+          );
+        },
+        () => []
+      );
+    },
+
+    writeMetadata(metadata: SerializedRequest[]) {
+      database$.then(
+        database => {
+          database
+            .transaction(METADATA_STORE_NAME, 'readonly')
```

Comment:
I don't know if you think it might be worth trying to share the 

``` js
database.transaction(...).objectStore(...)[method](...)
```
 logic? Might save a few bytes

Reply:
Let me test that :+1:

Reply:
That seems to increase the size by ~20B gzipped

Reply:
Nicely tested 👌